src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/MgmtConfigurationRecord.cs

```diff
@@ -335,7 +335,7 @@ internal ConfigurationSection FindImmediateParentSection(ConfigurationSection se
 
             string configKey = section.SectionInformation.SectionName;
             SectionRecord sectionRecord = GetSectionRecord(configKey, false);
-            if (sectionRecord.HasLocationInputs)
+            if ((sectionRecord != null) && sectionRecord.HasLocationInputs)
```

Comment:
This doesn't actually fix a null reference exception. _If_ sectionRecord could actually be null, this will cause the else block to be executed, which also immediately dereferences sectionRecord.

Unless there's a known repro for this causing a problem, let's revert this.

Reply:
Stefan, to correct the error, I propose below in line 346 to change the condition to <if ((sectionRecord != null) && sectionRecord.HasIndirectLocationInputs)>. Stefan, judging by the results of the static analysis, the value returned by the function GetSectionRecord(...) in the library System.Configuration.ConfigurationManager code base should be checked for NULL (you can search (sectionRecord==null or sectionRecord!=null) by searching in the files MgmtConfigurationRecord.cs and BaseConfigurationRecord.cs). I suggest you fix it. It is not possible to check for an error in this case using a test, because it was detected using a static analysis tool.

Reply:
This method is searching for immediate parent section. I assume that the immediate parent section is expected to always exist and that's why there is no null check. Maybe we can add Debug.Assert instead?

Reply:
Yes, let's try, just explain what needs to be done for this.